clustering/web/WFLY-13904_ProtoStream_support_for_distributed_web_applications.adoc

```diff
@@ -0,0 +1,102 @@
+= ProtoStream marshalling of web session attributes
+:author:            Paul Ferraro
+:email:             paul.ferraro@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Currently, WildFly uses JBoss Marshalling to marshal user objects (HttpSession attributes, SFSB instances, etc.) for the purposes of replication and persistence.
+
+ProtoStream (developed by the Infinispan team) is a Java API for marshalling Google Protocol buffers (i.e. Protobuf).
```

Comment:
Does it mean that ProtoStream as such is tested somewhere else? E.g. in Infinispan testsuite. Can it be mentioned in test plan? 

And we should in scope of this RFE focus on testing integration with WildFly?

Reply:
> Does it mean that ProtoStream as such is tested somewhere else? E.g. in Infinispan testsuite. Can it be mentioned in test plan?

ProtoStream has its own testsuite.  Infinispan has been using ProtoStream extensively since Infinispan 10.0.  Consequently, WF has been using ProtoStream to marshal Infinispan internals since WF20, and switched to ProtoStream for marshalling cache entries for our server, web, and ejb cache containers since WF21.

> And we should in scope of this RFE focus on testing integration with WildFly?

ProtoStream is already integrated into WildFly (since WF20).  This RFE makes it available specifically to marshal HttpSession attributes.